crates/sui-framework/packages/bridge/sources/message.move

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module bridge::message {
+    use std::vector;
+
+    use sui::bcs;
+
+    use bridge::message_types;
+
+    #[test_only]
+    use bridge::chain_ids;
+    #[test_only]
+    use bridge::treasury::token_id;
+    #[test_only]
+    use bridge::usdc::USDC;
+    #[test_only]
+    use sui::address;
+    #[test_only]
+    use sui::balance;
+    #[test_only]
+    use sui::coin;
+    #[test_only]
+    use sui::hex;
+    #[test_only]
+    use sui::test_scenario;
+
+    const CURRENT_MESSAGE_VERSION: u8 = 1;
+
+    struct BridgeMessage has copy, store, drop {
+        message_type: u8,
+        message_version: u8,
+        seq_num: u64,
+        payload: vector<u8>
+    }
+
+    struct BridgeMessageKey has copy, drop, store {
+        source_chain: u8,
+        message_type: u8,
+        bridge_seq_num: u64
+    }
+
+    struct TokenPayload has drop {
+        source_chain: u8,
+        sender_address: vector<u8>,
+        target_chain: u8,
+        target_address: vector<u8>,
+        token_type: u8,
```

Comment:
what extra overhead will it entail to store everything as u64? storage cost?

Reply:
If these are being stored (and never deleted) then it will make it a bit more costly.

Main thing is just wanting to make sure we don't get into a situation where further down the road we wished these had a larger range.

Reply:
storage cost is not that bad on Sui but might be expansive on ethereum? it might also be more expensive to process? Just guessing here, cc @Bridgerz for comment.

Reply:
Correct. Larger units consume more gas when both reading and writing from storage. Sorry for the delayed response!

Reply:
chatted with @Bridgerz, it will take a long time for us to exceed u8, so let's keep it as u8 for now.